compiler/test/suites/functions.re

```diff
@@ -385,4 +385,173 @@ truc()|},
     |},
     "which has a default value, but the matching argument does not.",
   );
+
+  assertRun(
+    "partial1",
+    {|
+      let f = (x, y, z) => { print(x); print(y); print(z) }
+      let pf = partial f(1, _, _)
+      pf(2, 3)
+    |},
+    "1\n2\n3\n",
+  );
+  assertRun(
+    "partial2",
+    {|
+      let f = (x, y, z) => { print(x); print(y); print(z) }
+      let pf = partial f(_, 2, _)
+      pf(1, 3)
+    |},
+    "1\n2\n3\n",
+  );
+  assertRun(
+    "partial3",
+    {|
+      let f = (x, y, z) => { print(x); print(y); print(z) }
+      let pf = partial f(_, x=1, _)
+      pf(2, 3)
+    |},
+    "1\n2\n3\n",
+  );
+  assertRun(
+    "partial4",
+    {|
+      let f = (x, y, z) => { print(x); print(y); print(z) }
+      let pf = partial f(_, x=1, _)
+      pf(2, 3)
+    |},
+    "1\n2\n3\n",
+  );
+  assertRun(
+    "partial5",
+    {|
+      let f = (x, y, z) => { print(x); print(y); print(z) }
+      let pf = partial f(_, x=1, _)
+      pf(z=3, y=2)
+    |},
+    "1\n2\n3\n",
+  );
+  assertRun(
+    "partial6",
+    {|
+      let f = (x, y, z) => { print(x); print(y); print(z) }
+      let pf = partial f(1, _, y=_)
+      pf(3, 2)
+    |},
+    "1\n2\n3\n",
```

Comment:
as a sanity check can we just add a `print("0")` between the partial application and the actual application of the function i think it's obvious that it will only call the function afterwards but I feel like we should still double check that.